RFE: improve usability by making logs more user friendly

#### Title
RFE: make logs more user friendly / usable

#### Description

Please consider the following suggestions to improve the logs' user friendliness. These suggestions come from me as well as from the customer I´ve been working with lately.

#### Actual Problem

- INFO level logs don´t always contain the information required, customers should be able to know easily that manifests or events are managed by CIS and if they are applied successfully or not.
- Some INFO level logs are clearly user friendly and should be considered DEBUG
- When an error occurs, and it is needed detailed information to troubleshoot it is, required to restart CIS to increase the DEBUG level.

See below specific examples & proposal

Many thanks

#### Solution Proposed

**A**- Every time there is an user or Kubernetes action, ie: create, update or delete a manifest there should be the following:

   1. A simple, user readable, message at INFO level indicating that CIS is taking care of the action, something alike:

    ```<cluster name> requested <action> in <resource type> <namespace>/<manifest name>```

  for example:

    ```ocp1 requested CREATE in Route test/web```
    ```ocp1 requested UPDATE in Service test/web-pool```

  2.  A corresponding message with INFO level indicating if the action has completed successfully or not:
  
    ```<cluster name> requested <action> in <resource type> <namespace>/<manifest name> <result><: optional error message>```

  for example:

    ```ocp1 requested CREATE in Route test/web resulted in SUCCESS```
    ```ocp1 requested UPDATE in Service test/web-pool resulted in FAILURE```

  Please consider that if the requests have some sort of CIS identifier, this could be prepended (ie: ```<transaction ID>``` to find the matching lines. Please also consider the capitalization of the key information action and result.

   if both item i and ii are excessive, at least item ii should be shown in the logs

**B** - I guess that some errors will not be discovered until the AS3 manifest will be submitted. I propose that:

  1. When a set of events trigger the creation of an updated AS3 manifest, a user friendly log message at INFO level is shown, for example:
  
  ```AS3: creating a new AS3 manifest for the next Kubernetes events```

  Then the messages mentioned at the top would follow.
  
  2. When the AS3 declaration is submitted a user readable log message shows the outcome of the AS3 submission, for example:

  ```AS3: posted AS3 manifest resulted in SUCCESS``` // with INFO log level

  ```AS3: posted AS3 manifest resulted in ERROR: <error message>``` // with ERROR log level


  Overall it could look as follows:

```
AS3: creating a new AS3 manifest for the next Kubernetes events
ocp1 requested CREATE in Route test/web
ocp1 requested UPDATE in Service test/web-pool
ocp1 requested CREATE in Route test/web resulted in SUCCESS
ocp1 requested UPDATE in Service test/web-pool resulted in SUCCESS
AS3: posted AS3 manifest resulted in SUCCESS
```
  I guess that if the one of the requested changes don´t success, then I understand that no AS3 declaration is submitted and something alike the following could be shown:

```
AS3: previous failures aborted posting an AS3 manifest
```
**C** -  The following log should not be ERROR but NOTICE level as it might be intentional to temporarily remove a Service from a Route or to do not have that Service in a given cluster.

```2023/10/16 14:20:50 [ERROR] [MultiCluster] Service {guestbook-app1 ocp2 eng-caas-guestbook-app1} not found! from cluster: ocp2```

In general I find that ERROR messages should be shown when there is a CIS problem which it inhibits working, ie: accessing an API or CIS HA not working properly, but not when the failure comes for the regular working of Kubernetes or AS3.

**D**-   if there is an AS3 error, then the AS3 declaration is dumped (like AS3DEBUG) regardless of the debug level. This way an issue can be troubleshooted without having to restart CIS

**E** -  include always cluster name in all messages, when not running in multi-cluster mode, cluster name could be a generic name, ie: "local"

**F** -  There are a set of messages with the ```[INFO] Text:``` token which should be changed to DEBUG level, they are not user readable

```
2023/10/16 13:08:09 [INFO] Text: '{"kind":"tm:sys:syscollectionstate","selfLink":"https://localhost/mgmt/tm/sys?ver=17.1.0.1","items":[{"reference":{"link":"https://localhost/mgmt/tm/sys/application?ver=17.1.0.1"
}},{"reference":{"link":"https://localhost/mgmt/tm/sys/crypto?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/daemon-log-settings?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/t
m/sys/diags?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/disk?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/dynad?ver=17.1.0.1"}},{"reference":{"link":"https://localho
st/mgmt/tm/sys/ecm?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/file?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/fpga?ver=17.1.0.1"}},{"reference":{"link":"https://l
ocalhost/mgmt/tm/sys/icall?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/ipfix?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/log-config?ver=17.1.0.1"}},{"reference":{"l
ink":"https://localhost/mgmt/tm/sys/pfman?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/sflow?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/software?ver=17.1.0.1"}},{"r
eference":{"link":"https://localhost/mgmt/tm/sys/turboflex?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/url-db?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/aom?ver=17
.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/autoscale-group?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/cluster?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mg
mt/tm/sys/config?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/core?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/daemon-ha?ver=17.1.0.1"}},{"reference":{"link":"https:
//localhost/mgmt/tm/sys/datastor?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/db?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/dns?ver=17.1.0.1"}},{"reference":{"link"
:"https://localhost/mgmt/tm/sys/feature-module?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/folder?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/global-settings?ver=17
.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/ha-group?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/httpd?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys
/iapp-restricted-key?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/icontrol-soap?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/internal-proxy?ver=17.1.0.1"}},{"referenc
e":{"link":"https://localhost/mgmt/tm/sys/log-rotate?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/management-dhcp?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/management-ip?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/management-ovsdb?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/management-proxy-config?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/management-route?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/ntp?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/outbound-smtp?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/provision?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/scriptd?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/service?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/smtp-server?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/snmp?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/sshd?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/state-mirroring?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/syslog?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/telemd?ver=17.1.0.1"}},{"reference":{"link":"https://localhost/mgmt/tm/sys/ucs?ver=17.1.0.1"}}]}'

2023/10/16 13:08:09 [INFO] Enqueueing Namespace: &Namespace{ObjectMeta:{cis-mc-onetier    a37b8e45-f02c-4b4c-b1a7-a286feed4dcd 10160835 0 2023-10-16 13:07:55 +0000 UTC <nil> <nil> map[environment:test kubernetes.io/metadata.name:cis-mc-onetier pod-security.kubernetes.io/audit:privileged pod-security.kubernetes.io/audit-version:v1.24 pod-security.kubernetes.io/warn:privileged pod-security.kubernetes.io/warn-version:v1.24] map[openshift.io/sa.scc.mcs:s0:c29,c24 openshift.io/sa.scc.supplemental-groups:1000860000/10000 openshift.io/sa.scc.uid-range:1000860000/10000] [] []  [{kubectl-create Update v1 2023-10-16 13:07:55 +0000 UTC FieldsV1 {"f:metadata":{"f:labels":{".":{},"f:kubernetes.io/metadata.name":{}}}}} {cluster-policy-controller Update v1 2023-10-16 13:07:57 +0000 UTC FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:openshift.io/sa.scc.mcs":{},"f:openshift.io/sa.scc.supplemental-groups":{},"f:openshift.io/sa.scc.uid-range":{}},"f:labels":{"f:pod-security.kubernetes.io/audit":{},"f:pod-security.kubernetes.io/audit-version":{},"f:pod-security.kubernetes.io/warn":{},"f:pod-security.kubernetes.io/warn-version":{}}}}} {kubectl-label Update v1 2023-10-16 13:08:03 +0000 UTC FieldsV1 {"f:metadata":{"f:labels":{"f:environment":{}}}}}]},Spec:NamespaceSpec{Finalizers:[kubernetes],},Status:NamespaceStatus{Phase:Active,Conditions:[]NamespaceCondition{},},}

2023/10/16 13:08:09 [INFO] Text: '{"kind":"tm:net:fdb:tunnel:records:recordscollectionstate","selfLink":"https://localhost/mgmt/tm/net/fdb/tunnel/~Common~http-tunnel/records?ver=17.1.0.1","items":[]}'
2023/10/16 13:08:09 [INFO] Text: '{"kind":"tm:net:fdb:tunnel:records:recordscollectionstate","selfLink":"https://localhost/mgmt/tm/net/fdb/tunnel/~Common~socks-tunnel/records?ver=17.1.0.1","items":[]}'

2023/10/16 13:08:40 [INFO] Text: '{"kind":"tm:auth:partition:partitionstate","name":"Common","fullPath":"Common","generation":1,"selfLink":"https://localhost/mgmt/tm/auth/partition/Common?ver=17.1.0.1","defaultRouteDomain":0,"description":"Updated by AS3 at Fri, 13 Oct 2023 14:27:49 GMT"}'
```

**G** -  Additionally, the next ones are additionally not expected when using OVN in routed mode

2023/10/16 13:08:40 [INFO] Text: '{"kind":"tm:auth:partition:partitionstate","name":"Common","fullPath":"Common","generation":1,"selfLink":"https://localhost/mgmt/tm/auth/partition/Common?ver=17.1.0.1","defaultRouteDomain":0,"description":"Updated by AS3 at Fri, 13 Oct 2023 14:27:49 GMT"}'

--- a/cmd/k8s-bigip-ctlr/main.go
+++ b/cmd/k8s-bigip-ctlr/main.go
@@ -581,7 +581,7 @@ func verifyArgs() error {
 	if *multiClusterMode != "standalone" && *multiClusterMode != "primary" && *multiClusterMode != "secondary" && *multiClusterMode != "" {
 		return fmt.Errorf("'%v' is not a valid multi cluster mode, allowed values are: standalone/primary/secondary", *multiClusterMode)
 	} else if *multiClusterMode != "" {
-		log.Debugf("[MultiCluster] CIS running with multi-cluster-mode: %s", *multiClusterMode)
+		log.Infof("[MultiCluster] CIS running with multi-cluster-mode: %s", *multiClusterMode)
 	}
 
 	if (len(*routeSpecConfigmap) == 0 && len(*extendedSpecConfigmap) == 0) && *multiClusterMode != "" {

--- a/f5-bigip-ctlr-operator/bundle/manifests/f5-bigip-ctlr-operator.clusterserviceversion.yaml
+++ b/f5-bigip-ctlr-operator/bundle/manifests/f5-bigip-ctlr-operator.clusterserviceversion.yaml
@@ -115,7 +115,10 @@ spec:
     ### Support for CIS CustomResources
     Install CRDs manually as follows if using CIS CustomResources (VirtualServer/TransportServer/IngressLink):
     ```
-    oc apply -f https://raw.githubusercontent.com/F5Networks/k8s-bigip-ctlr/master/docs/config_examples/crd/Install/customresourcedefinitions.yml
+    export CIS_VERSION=<cis-version>
+    # For example
+    # export CIS_VERSION=v2.12.0
+    oc create -f https://raw.githubusercontent.com/F5Networks/k8s-bigip-ctlr/${CIS_VERSION}/docs/config_examples/customResourceDefinitions/customresourcedefinitions.yml
     ```
   displayName: F5 Container Ingress Services
   icon:

--- a/pkg/controller/backend.go
+++ b/pkg/controller/backend.go
@@ -242,6 +242,7 @@ func (agent *Agent) agentWorker() {
 		// Fetch the latest config from channel
 		select {
 		case rsConfig = <-agent.postChan:
+			log.Infof("%v[AS3] Processing request", getRequestPrefix(rsConfig.reqId))
 		case <-time.After(1 * time.Microsecond):
 		}
 
@@ -252,10 +253,11 @@ func (agent *Agent) agentWorker() {
 		if agent.GTMPostManager != nil {
 			agent.GTMPostManager.PostGTMConfig(rsConfig)
 		}
-
+		log.Infof("%v[AS3] creating a new AS3 manifest", getRequestPrefix(rsConfig.reqId))
 		decl := agent.createTenantAS3Declaration(rsConfig)
 
 		if len(agent.incomingTenantDeclMap) == 0 {
+			log.Infof("%v[AS3] No tenants found in request", getRequestPrefix(rsConfig.reqId))
 			agent.declUpdate.Unlock()
 			continue
 		}

--- a/pkg/controller/controller.go
+++ b/pkg/controller/controller.go
@@ -83,6 +83,8 @@ const (
 	NodePort = "nodeport"
 	Cluster  = "cluster"
 
+	Local = "local"
+
 	StandAloneCIS = "standalone"
 	SecondaryCIS  = "secondary"
 	PrimaryCIS    = "primary"

--- a/pkg/controller/gtmBackend.go
+++ b/pkg/controller/gtmBackend.go
@@ -34,10 +34,12 @@ func (agent *Agent) gtmWorker() {
 		// Fetch the latest config from channel
 		select {
 		case rsConfig = <-agent.GTMPostManager.PostManager.postChan:
+			log.Infof("%v[AS3] Processing request", getRequestPrefix(rsConfig.reqId))
 		case <-time.After(1 * time.Microsecond):
 		}
 		adc := as3ADC{}
 		agent.GTMPostManager.incomingTenantDeclMap = make(map[string]as3Tenant)
+		log.Infof("%v[AS3] creating a new AS3 manifest", getRequestPrefix(rsConfig.reqId))
 		for tenant, cfg := range agent.createAS3GTMConfigADC(rsConfig, adc) {
 			if !reflect.DeepEqual(cfg, agent.GTMPostManager.cachedTenantDeclMap[tenant]) {
 				agent.GTMPostManager.incomingTenantDeclMap[tenant] = cfg.(as3Tenant)
@@ -50,6 +52,7 @@ func (agent *Agent) gtmWorker() {
 		}
 
 		if len(agent.GTMPostManager.incomingTenantDeclMap) == 0 {
+			log.Infof("%v[AS3] No tenants found in request", getRequestPrefix(rsConfig.reqId))
 			agent.declUpdate.Unlock()
 			continue
 		}

--- a/pkg/controller/informers.go
+++ b/pkg/controller/informers.go
@@ -19,6 +19,8 @@ package controller
 import (
 	"context"
 	"fmt"
+	"io"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/client-go/rest"
 	"reflect"
 	"time"
@@ -32,6 +34,7 @@ import (
 	cisinfv1 "github.com/F5Networks/k8s-bigip-ctlr/v2/config/client/informers/externalversions/cis/v1"
 	log "github.com/F5Networks/k8s-bigip-ctlr/v2/pkg/vlogger"
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/tools/cache"
 )
@@ -131,7 +134,7 @@ func (crInfr *CRInformer) start() {
 		cacheSyncs = append(cacheSyncs, crInfr.ilInformer.HasSynced)
 	}
 	cache.WaitForNamedCacheSync(
-		"F5 CIS CRD Controller",
+		"F5 CIS Ingress Controller",
 		crInfr.stopCh,
 		cacheSyncs...,
 	)
@@ -437,6 +440,7 @@ func (ctlr *Controller) addNodeEventUpdateHandler(nodeInformer *NodeInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.SetupNodeProcessing(nodeInformer.clusterName) },
 			},
 		)
+		nodeInformer.nodeInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(NodeUpdate, nodeInformer.clusterName))
 	}
 }
 
@@ -553,6 +557,7 @@ func (ctlr *Controller) addCustomResourceEventHandlers(crInf *CRInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueDeletedVirtualServer(obj) },
 			},
 		)
+		crInf.vsInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(VirtualServer, Local))
 	}
 
 	if crInf.tlsInformer != nil {
@@ -563,6 +568,7 @@ func (ctlr *Controller) addCustomResourceEventHandlers(crInf *CRInformer) {
 				// DeleteFunc: func(obj interface{}) { ctlr.enqueueTLSProfile(obj) },
 			},
 		)
+		crInf.tlsInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(TLSProfile, Local))
 	}
 
 	if crInf.tsInformer != nil {
@@ -573,6 +579,7 @@ func (ctlr *Controller) addCustomResourceEventHandlers(crInf *CRInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueDeletedTransportServer(obj) },
 			},
 		)
+		crInf.tsInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(TransportServer, Local))
 	}
 
 	if crInf.ilInformer != nil {
@@ -583,6 +590,7 @@ func (ctlr *Controller) addCustomResourceEventHandlers(crInf *CRInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueDeletedIngressLink(obj) },
 			},
 		)
+		crInf.ilInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(IngressLink, Local))
 	}
 }
 
@@ -595,6 +603,7 @@ func (ctlr *Controller) addCommonResourceEventHandlers(comInf *CommonInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueDeletedService(obj, "") },
 			},
 		)
+		comInf.svcInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(Service, Local))
 	}
 
 	if comInf.epsInformer != nil {
@@ -605,6 +614,7 @@ func (ctlr *Controller) addCommonResourceEventHandlers(comInf *CommonInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueEndpoints(obj, Delete, "") },
 			},
 		)
+		comInf.epsInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(Endpoints, Local))
 	}
 
 	if comInf.ednsInformer != nil {
@@ -614,6 +624,7 @@ func (ctlr *Controller) addCommonResourceEventHandlers(comInf *CommonInformer) {
 				UpdateFunc: func(oldObj, newObj interface{}) { ctlr.enqueueUpdatedExternalDNS(oldObj, newObj) },
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueDeletedExternalDNS(obj) },
 			})
+		comInf.ednsInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(ExternalDNS, Local))
 	}
 
 	if comInf.plcInformer != nil {
@@ -624,6 +635,7 @@ func (ctlr *Controller) addCommonResourceEventHandlers(comInf *CommonInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueDeletedPolicy(obj) },
 			},
 		)
+		comInf.plcInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(CustomPolicy, Local))
 	}
 
 	if comInf.podInformer != nil {
@@ -634,6 +646,7 @@ func (ctlr *Controller) addCommonResourceEventHandlers(comInf *CommonInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueDeletedPod(obj, "") },
 			},
 		)
+		comInf.podInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(Pod, Local))
 	}
 
 	if comInf.secretsInformer != nil {
@@ -644,6 +657,7 @@ func (ctlr *Controller) addCommonResourceEventHandlers(comInf *CommonInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueSecret(obj, Delete) },
 			},
 		)
+		comInf.secretsInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(Secret, Local))
 	}
 
 	if comInf.cmInformer != nil {
@@ -654,6 +668,7 @@ func (ctlr *Controller) addCommonResourceEventHandlers(comInf *CommonInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueDeletedConfigmap(obj) },
 			},
 		)
+		comInf.cmInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(ConfigMap, Local))
 	}
 
 }
@@ -667,6 +682,7 @@ func (ctlr *Controller) addNativeResourceEventHandlers(nrInf *NRInformer) {
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueRoute(obj, Delete) },
 			},
 		)
+		nrInf.routeInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(Route, Local))
 	}
 }
 
@@ -685,7 +701,7 @@ func (ctlr *Controller) enqueueIPAM(obj interface{}) {
 		return
 	}
 
-	log.Infof("Enqueueing IPAM: %v", ipamObj)
+	log.Debugf("Enqueueing IPAM: %v", ipamObj)
 	key := &rqKey{
 		namespace: ipamObj.ObjectMeta.Namespace,
 		kind:      IPAM,
@@ -709,7 +725,7 @@ func (ctlr *Controller) enqueueUpdatedIPAM(oldObj, newObj interface{}) {
 		return
 	}
 
-	log.Infof("Enqueueing Updated IPAM: %v", curIpam)
+	log.Debugf("Enqueueing Updated IPAM: %v", curIpam)
 	key := &rqKey{
 		namespace: curIpam.ObjectMeta.Namespace,
 		kind:      IPAM,
@@ -728,7 +744,7 @@ func (ctlr *Controller) enqueueDeletedIPAM(obj interface{}) {
 		return
 	}
 
-	log.Infof("Enqueueing IPAM: %v", ipamObj)
+	log.Debugf("Enqueueing IPAM: %v", ipamObj)
 	key := &rqKey{
 		namespace: ipamObj.ObjectMeta.Namespace,
 		kind:      IPAM,
@@ -820,7 +836,7 @@ func (ctlr *Controller) enqueueDeletedVirtualServer(obj interface{}) {
 
 func (ctlr *Controller) enqueueTLSProfile(obj interface{}, event string) {
 	tls := obj.(*cisapiv1.TLSProfile)
-	log.Infof("Enqueueing TLSProfile: %v", tls)
+	log.Debugf("Enqueueing TLSProfile: %v", tls)
 	key := &rqKey{
 		namespace: tls.ObjectMeta.Namespace,
 		kind:      TLSProfile,
@@ -834,7 +850,7 @@ func (ctlr *Controller) enqueueTLSProfile(obj interface{}, event string) {
 
 func (ctlr *Controller) enqueueTransportServer(obj interface{}) {
 	ts := obj.(*cisapiv1.TransportServer)
-	log.Infof("Enqueueing TransportServer: %v", ts)
+	log.Debugf("Enqueueing TransportServer: %v", ts)
 	key := &rqKey{
 		namespace: ts.ObjectMeta.Namespace,
 		kind:      TransportServer,
@@ -910,7 +926,7 @@ func (ctlr *Controller) enqueueDeletedTransportServer(obj interface{}) {
 
 func (ctlr *Controller) enqueuePolicy(obj interface{}, event string) {
 	pol := obj.(*cisapiv1.Policy)
-	log.Infof("Enqueueing Policy: %v", pol)
+	log.Debugf("Enqueueing Policy: %v", pol)
 	key := &rqKey{
 		namespace: pol.ObjectMeta.Namespace,
 		kind:      CustomPolicy,
@@ -924,7 +940,7 @@ func (ctlr *Controller) enqueuePolicy(obj interface{}, event string) {
 
 func (ctlr *Controller) enqueueDeletedPolicy(obj interface{}) {
 	pol := obj.(*cisapiv1.Policy)
-	log.Infof("Enqueueing Policy: %v", pol)
+	log.Debugf("Enqueueing Policy: %v", pol)
 	key := &rqKey{
 		namespace: pol.ObjectMeta.Namespace,
 		kind:      CustomPolicy,
@@ -938,7 +954,7 @@ func (ctlr *Controller) enqueueDeletedPolicy(obj interface{}) {
 
 func (ctlr *Controller) enqueueIngressLink(obj interface{}) {
 	ingLink := obj.(*cisapiv1.IngressLink)
-	log.Infof("Enqueueing IngressLink: %v", ingLink)
+	log.Debugf("Enqueueing IngressLink: %v", ingLink)
 	key := &rqKey{
 		namespace: ingLink.ObjectMeta.Namespace,
 		kind:      IngressLink,
@@ -952,7 +968,7 @@ func (ctlr *Controller) enqueueIngressLink(obj interface{}) {
 
 func (ctlr *Controller) enqueueDeletedIngressLink(obj interface{}) {
 	ingLink := obj.(*cisapiv1.IngressLink)
-	log.Infof("Enqueueing IngressLink: %v on Delete", ingLink)
+	log.Debugf("Enqueueing IngressLink: %v on Delete", ingLink)
 	key := &rqKey{
 		namespace: ingLink.ObjectMeta.Namespace,
 		kind:      IngressLink,
@@ -990,7 +1006,7 @@ func (ctlr *Controller) enqueueUpdatedIngressLink(oldObj, newObj interface{}) {
 		ctlr.resourceQueue.Add(key)
 	}
 
-	log.Infof("Enqueueing IngressLink: %v on Update", newIngLink)
+	log.Debugf("Enqueueing IngressLink: %v on Update", newIngLink)
 	key := &rqKey{
 		namespace: newIngLink.ObjectMeta.Namespace,
 		kind:      IngressLink,
@@ -1004,7 +1020,7 @@ func (ctlr *Controller) enqueueUpdatedIngressLink(oldObj, newObj interface{}) {
 
 func (ctlr *Controller) enqueueExternalDNS(obj interface{}) {
 	edns := obj.(*cisapiv1.ExternalDNS)
-	log.Infof("Enqueueing ExternalDNS: %v", edns)
+	log.Debugf("Enqueueing ExternalDNS: %v", edns)
 	key := &rqKey{
 		namespace: edns.ObjectMeta.Namespace,
 		kind:      ExternalDNS,
@@ -1032,7 +1048,7 @@ func (ctlr *Controller) enqueueUpdatedExternalDNS(oldObj, newObj interface{}) {
 		ctlr.resourceQueue.Add(key)
 	}
 
-	log.Infof("Enqueueing Updated ExternalDNS: %v", edns)
+	log.Debugf("Enqueueing Updated ExternalDNS: %v", edns)
 	key := &rqKey{
 		namespace: edns.ObjectMeta.Namespace,
 		kind:      ExternalDNS,
@@ -1046,7 +1062,7 @@ func (ctlr *Controller) enqueueUpdatedExternalDNS(oldObj, newObj interface{}) {
 
 func (ctlr *Controller) enqueueDeletedExternalDNS(obj interface{}) {
 	edns := obj.(*cisapiv1.ExternalDNS)
-	log.Infof("Enqueueing ExternalDNS: %v", edns)
+	log.Debugf("Enqueueing ExternalDNS: %v", edns)
 	key := &rqKey{
 		namespace: edns.ObjectMeta.Namespace,
 		kind:      ExternalDNS,
@@ -1391,7 +1407,7 @@ func (ctlr *Controller) createNamespaceLabeledInformer(label string) error {
 
 func (ctlr *Controller) enqueueNamespace(obj interface{}) {
 	ns := obj.(*corev1.Namespace)
-	log.Infof("Enqueueing Namespace: %v", ns)
+	log.Debugf("Enqueueing Namespace: %v", ns)
 	key := &rqKey{
 		namespace: ns.ObjectMeta.Namespace,
 		kind:      Namespace,
@@ -1404,7 +1420,7 @@ func (ctlr *Controller) enqueueNamespace(obj interface{}) {
 
 func (ctlr *Controller) enqueueDeletedNamespace(obj interface{}) {
 	ns := obj.(*corev1.Namespace)
-	log.Infof("Enqueueing Namespace: %v on Delete", ns)
+	log.Debugf("Enqueueing Namespace: %v on Delete", ns)
 	key := &rqKey{
 		namespace: ns.ObjectMeta.Namespace,
 		kind:      Namespace,
@@ -1435,4 +1451,22 @@ func (ctlr *Controller) enqueuePrimaryClusterProbeEvent() {
 		kind: HACIS,
 	}
 	ctlr.resourceQueue.Add(key)
+}
+
+func (ctlr *Controller) getErrorHandlerFunc(rsType, clusterName string) func(r *cache.Reflector, err error) {
+	return func(r *cache.Reflector, err error) {
+		switch {
+		case apierrors.IsResourceExpired(err) || apierrors.IsGone(err):
+			// Don't set LastSyncResourceVersionUnavailable - LIST call with ResourceVersion=RV already
+			// has a semantic that it returns data at least as fresh as provided RV.
+			// So first try to LIST with setting RV to resource version of last observed object.
+			log.Errorf("Watch of %v in cluster %v closed with: %v", rsType, clusterName, err)
+		case err == io.EOF:
+			// watch closed normally
+		case err == io.ErrUnexpectedEOF:
+			log.Errorf("Watch for %v in cluster %v closed with unexpected EOF: %v", rsType, clusterName, err)
+		default:
+			utilruntime.HandleError(fmt.Errorf("[ERROR] Failed to watch %v in cluster %v: %v", rsType, clusterName, err))
+		}
+	}
 }

--- a/pkg/controller/multiClusterInformers.go
+++ b/pkg/controller/multiClusterInformers.go
@@ -181,6 +181,7 @@ func (ctlr *Controller) addMultiClusterPoolEventHandlers(poolInf *MultiClusterPo
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueDeletedService(obj, poolInf.clusterName) },
 			},
 		)
+		poolInf.svcInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(Service, poolInf.clusterName))
 	}
 
 	if poolInf.epsInformer != nil {
@@ -191,6 +192,7 @@ func (ctlr *Controller) addMultiClusterPoolEventHandlers(poolInf *MultiClusterPo
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueEndpoints(obj, Delete, poolInf.clusterName) },
 			},
 		)
+		poolInf.epsInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(Endpoints, poolInf.clusterName))
 	}
 	if poolInf.podInformer != nil {
 		poolInf.podInformer.AddEventHandler(
@@ -200,6 +202,7 @@ func (ctlr *Controller) addMultiClusterPoolEventHandlers(poolInf *MultiClusterPo
 				DeleteFunc: func(obj interface{}) { ctlr.enqueueDeletedPod(obj, poolInf.clusterName) },
 			},
 		)
+		poolInf.podInformer.SetWatchErrorHandler(ctlr.getErrorHandlerFunc(Pod, poolInf.clusterName))
 	}
 }
 

--- a/pkg/controller/nativeResourceWorker.go
+++ b/pkg/controller/nativeResourceWorker.go
@@ -976,7 +976,7 @@ func (ctlr *Controller) setNamespaceLabelMode(cm *v1.ConfigMap) error {
 						}
 					}
 				} else {
-					log.Debugf("%v Added namespace label informer: %v", ctlr.getMultiClusterLog(), nsLabel)
+					log.Infof("%v Added namespace label informer: %v", ctlr.getMultiClusterLog(), nsLabel)
 					ctlr.nsInformers[nsLabel].start()
 				}
 			}

--- a/pkg/controller/postManager.go
+++ b/pkg/controller/postManager.go
@@ -22,7 +22,7 @@ import (
 	"crypto/x509"
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
+	"io"
 	"net/http"
 	"strconv"
 	"strings"
@@ -126,10 +126,11 @@ func (postMgr *PostManager) postConfig(cfg *agentConfig) {
 	httpReqBody := bytes.NewBuffer([]byte(cfg.data))
 	req, err := http.NewRequest("POST", cfg.as3APIURL, httpReqBody)
 	if err != nil {
-		log.Errorf("[AS3]%v Creating new HTTP request error: %v ", postMgr.postManagerPrefix, err)
+		log.Errorf("%v[AS3]%v Creating new HTTP request error: %v ", getRequestPrefix(cfg.id), postMgr.postManagerPrefix, err)
 		return
 	}
 	log.Debugf("[AS3]%v posting request to %v", postMgr.postManagerPrefix, cfg.as3APIURL)
+	log.Infof("%v[AS3]%v posting request to %v for %v tenants", getRequestPrefix(cfg.id), postMgr.postManagerPrefix, postMgr.BIGIPURL, getTenantsFromUri(cfg.as3APIURL))
 	req.SetBasicAuth(postMgr.BIGIPUsername, postMgr.BIGIPPassword)
 
 	httpResp, responseMap := postMgr.httpPOST(req)
@@ -143,17 +144,23 @@ func (postMgr *PostManager) postConfig(cfg *agentConfig) {
 
 	switch httpResp.StatusCode {
 	case http.StatusOK:
+		log.Infof("%v[AS3]%v post resulted in SUCCESS", getRequestPrefix(cfg.id), postMgr.postManagerPrefix)
 		postMgr.handleResponseStatusOK(responseMap)
 	case http.StatusCreated, http.StatusAccepted:
+		log.Infof("%v[AS3]%v post resulted in ACCEPTED", getRequestPrefix(cfg.id), postMgr.postManagerPrefix)
 		postMgr.handleResponseAccepted(responseMap)
 	case http.StatusMultiStatus:
-		postMgr.handleMultiStatus(responseMap)
+		log.Infof("%v[AS3]%v post resulted in MULTI-STATUS", getRequestPrefix(cfg.id), postMgr.postManagerPrefix)
+		postMgr.handleMultiStatus(responseMap, cfg.id)
 	case http.StatusServiceUnavailable:
-		postMgr.handleResponseStatusServiceUnavailable(responseMap)
+		log.Infof("%v[AS3]%v post resulted in RETRY", getRequestPrefix(cfg.id), postMgr.postManagerPrefix)
+		postMgr.handleResponseStatusServiceUnavailable(responseMap, cfg.id)
 	case http.StatusNotFound:
-		postMgr.handleResponseStatusNotFound(responseMap)
+		log.Infof("%v[AS3]%v post resulted in FAILURE", getRequestPrefix(cfg.id), postMgr.postManagerPrefix)
+		postMgr.handleResponseStatusNotFound(responseMap, cfg.id)
 	default:
-		postMgr.handleResponseOthers(responseMap, cfg)
+		log.Infof("%v[AS3]%v post resulted in FAILURE", getRequestPrefix(cfg.id), postMgr.postManagerPrefix)
+		postMgr.handleResponseOthers(responseMap, cfg.id)
 	}
 }
 
@@ -174,7 +181,7 @@ func (postMgr *PostManager) httpPOST(request *http.Request) (*http.Response, map
 	}
 	defer httpResp.Body.Close()
 
-	body, err := ioutil.ReadAll(httpResp.Body)
+	body, err := io.ReadAll(httpResp.Body)
 	if err != nil {
 		log.Errorf("[AS3]%v REST call response error: %v ", postMgr.postManagerPrefix, err)
 		return nil, nil
@@ -242,6 +249,11 @@ func (postMgr *PostManager) getTenantConfigStatus(id string) {
 				} else {
 					log.Debugf("[AS3]%v Response from BIG-IP: code: %v --- tenant:%v --- message: %v", postMgr.postManagerPrefix, v["code"], v["tenant"], v["message"])
 				}
+				intId, err := strconv.Atoi(id)
+				if err == nil {
+					log.Infof("%v[AS3]%v post resulted in SUCCESS", getRequestPrefix(intId), postMgr.postManagerPrefix)
+				}
+
 			}
 		}
 	} else if httpResp.StatusCode != http.StatusServiceUnavailable {
@@ -250,7 +262,7 @@ func (postMgr *PostManager) getTenantConfigStatus(id string) {
 	}
 }
 
-func (postMgr *PostManager) handleMultiStatus(responseMap map[string]interface{}) {
+func (postMgr *PostManager) handleMultiStatus(responseMap map[string]interface{}, id int) {
 	if results, ok := (responseMap["results"]).([]interface{}); ok {
 		declaration := (responseMap["declaration"]).(interface{}).(map[string]interface{})
 		for _, value := range results {
@@ -258,7 +270,7 @@ func (postMgr *PostManager) handleMultiStatus(responseMap map[string]interface{}
 
 			if v["code"].(float64) != 200 {
 				postMgr.updateTenantResponseCode(int(v["code"].(float64)), "", v["tenant"].(string), false)
-				log.Errorf("[AS3]%v Error response from BIG-IP: code: %v --- tenant:%v --- message: %v", postMgr.postManagerPrefix, v["code"], v["tenant"], v["message"])
+				log.Errorf("%v[AS3]%v Error response from BIG-IP: code: %v --- tenant:%v --- message: %v", getRequestPrefix(id), postMgr.postManagerPrefix, v["code"], v["tenant"], v["message"])
 			} else {
 				postMgr.updateTenantResponseCode(int(v["code"].(float64)), "", v["tenant"].(string), updateTenantDeletion(v["tenant"].(string), declaration))
 				log.Debugf("[AS3]%v Response from BIG-IP: code: %v --- tenant:%v --- message: %v", postMgr.postManagerPrefix, v["code"], v["tenant"], v["message"])
@@ -275,19 +287,19 @@ func (postMgr *PostManager) handleResponseAccepted(responseMap map[string]interf
 	}
 }
 
-func (postMgr *PostManager) handleResponseStatusServiceUnavailable(responseMap map[string]interface{}) {
+func (postMgr *PostManager) handleResponseStatusServiceUnavailable(responseMap map[string]interface{}, id int) {
 	if err, ok := (responseMap["error"]).(map[string]interface{}); ok {
-		log.Errorf("[AS3]%v Big-IP Responded with error code: %v", postMgr.postManagerPrefix, err["code"])
+		log.Errorf("%v[AS3]%v Big-IP Responded with error code: %v", getRequestPrefix(id), postMgr.postManagerPrefix, err["code"])
 	}
 	log.Debugf("[AS3]%v Response from BIG-IP: BIG-IP is busy, waiting %v seconds and re-posting the declaration", postMgr.postManagerPrefix, timeoutMedium)
 	postMgr.updateTenantResponseCode(http.StatusServiceUnavailable, "", "", false)
 }
 
-func (postMgr *PostManager) handleResponseStatusNotFound(responseMap map[string]interface{}) {
+func (postMgr *PostManager) handleResponseStatusNotFound(responseMap map[string]interface{}, id int) {
 	if err, ok := (responseMap["error"]).(map[string]interface{}); ok {
-		log.Errorf("[AS3]%v Big-IP Responded with error code: %v", postMgr.postManagerPrefix, err["code"])
+		log.Errorf("%v[AS3]%v Big-IP Responded with error code: %v", getRequestPrefix(id), postMgr.postManagerPrefix, err["code"])
 	} else {
-		log.Errorf("[AS3]%v Big-IP Responded with error code: %v", postMgr.postManagerPrefix, http.StatusNotFound)
+		log.Errorf("%v[AS3]%v Big-IP Responded with error code: %v", getRequestPrefix(id), postMgr.postManagerPrefix, http.StatusNotFound)
 	}
 	if postMgr.LogAS3Response {
 		postMgr.logAS3Response(responseMap)
@@ -295,21 +307,21 @@ func (postMgr *PostManager) handleResponseStatusNotFound(responseMap map[string]
 	postMgr.updateTenantResponseCode(http.StatusNotFound, "", "", false)
 }
 
-func (postMgr *PostManager) handleResponseOthers(responseMap map[string]interface{}, cfg *agentConfig) {
+func (postMgr *PostManager) handleResponseOthers(responseMap map[string]interface{}, id int) {
 	if postMgr.LogAS3Response {
 		postMgr.logAS3Response(responseMap)
 	}
 	if results, ok := (responseMap["results"]).([]interface{}); ok {
 		for _, value := range results {
 			v := value.(map[string]interface{})
-			log.Errorf("[AS3]%v Response from BIG-IP: code: %v --- tenant:%v --- message: %v", postMgr.postManagerPrefix, v["code"], v["tenant"], v["message"])
+			log.Errorf("%v[AS3]%v Response from BIG-IP: code: %v --- tenant:%v --- message: %v", getRequestPrefix(id), postMgr.postManagerPrefix, v["code"], v["tenant"], v["message"])
 			postMgr.updateTenantResponseCode(int(v["code"].(float64)), "", v["tenant"].(string), false)
 		}
 	} else if err, ok := (responseMap["error"]).(map[string]interface{}); ok {
-		log.Errorf("[AS3]%v Big-IP Responded with error code: %v", postMgr.postManagerPrefix, err["code"])
+		log.Errorf("%v[AS3]%v Big-IP Responded with error code: %v", getRequestPrefix(id), postMgr.postManagerPrefix, err["code"])
 		postMgr.updateTenantResponseCode(int(err["code"].(float64)), "", "", false)
 	} else {
-		log.Errorf("[AS3]%v Big-IP Responded with code: %v", postMgr.postManagerPrefix, responseMap["code"])
+		log.Errorf("%v[AS3]%v Big-IP Responded with code: %v", getRequestPrefix(id), postMgr.postManagerPrefix, responseMap["code"])
 		postMgr.updateTenantResponseCode(int(responseMap["code"].(float64)), "", "", false)
 	}
 }
@@ -419,7 +431,7 @@ func (postMgr *PostManager) httpReq(request *http.Request) (*http.Response, map[
 	}
 	defer httpResp.Body.Close()
 
-	body, err := ioutil.ReadAll(httpResp.Body)
+	body, err := io.ReadAll(httpResp.Body)
 	if err != nil {
 		log.Errorf("[AS3]%v REST call response error: %v ", postMgr.postManagerPrefix, err)
 		return nil, nil
@@ -693,4 +705,21 @@ func (postMgr *PostManager) retryFailedTenant(userAgent string) {
 		postMgr.updateTenantResponseMap(false)
 	}
 
+}
+
+// function for returning the prefix string for request id
+func getRequestPrefix(id int) string {
+	if id == 0 {
+		return "[Retry]"
+	}
+	return fmt.Sprintf("[Request: %v]", id)
+}
+
+// function for returning the tenants from URI
+func getTenantsFromUri(uri string) string {
+	res := strings.Split(uri, "declare/")
+	if len(res[0]) == 0 {
+		return "all"
+	}
+	return res[1]
 }

--- a/pkg/controller/pythonDriver.go
+++ b/pkg/controller/pythonDriver.go
@@ -114,8 +114,13 @@ func runBigIPDriver(pid chan<- int, cmd *exec.Cmd) {
 					log.Error(scanOut.Text())
 				} else if strings.Contains(scanOut.Text(), "CRITICAL]") {
 					log.Critical(scanOut.Text())
-				} else {
+				} else if strings.Contains(scanOut.Text(), "INFO]") && (strings.Contains(scanOut.Text(), "Creating") ||
+					strings.Contains(scanOut.Text(), "Updating") ||
+					strings.Contains(scanOut.Text(), "Deleting")) {
 					log.Info(scanOut.Text())
+				} else {
+					// moving cccl info level logs to debug
+					log.Debug(scanOut.Text())
 				}
 			} else {
 				break

--- a/pkg/controller/worker.go
+++ b/pkg/controller/worker.go
@@ -260,7 +260,7 @@ func (ctlr *Controller) processResources() bool {
 			err := ctlr.processRoutes(routeGroup, false)
 			if err != nil {
 				// TODO
-				utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+				utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 				isRetryableError = true
 			}
 		}
@@ -306,7 +306,7 @@ func (ctlr *Controller) processResources() bool {
 		err := ctlr.processVirtualServers(virtual, rscDelete)
 		if err != nil {
 			// TODO
-			utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+			utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 			isRetryableError = true
 		}
 		if rKey.event != Create && ctlr.multiClusterMode != "" {
@@ -326,7 +326,7 @@ func (ctlr *Controller) processResources() bool {
 			err := ctlr.processVirtualServers(virtual, false)
 			if err != nil {
 				// TODO
-				utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+				utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 				isRetryableError = true
 			}
 		}
@@ -359,7 +359,7 @@ func (ctlr *Controller) processResources() bool {
 					err := ctlr.processVirtualServers(virtual, false)
 					if err != nil {
 						// TODO
-						utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+						utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 						isRetryableError = true
 					}
 				}
@@ -393,7 +393,7 @@ func (ctlr *Controller) processResources() bool {
 		err := ctlr.processTransportServers(virtual, rscDelete)
 		if err != nil {
 			// TODO
-			utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+			utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 			isRetryableError = true
 		}
 		if rKey.event != Create && ctlr.multiClusterMode != "" {
@@ -418,7 +418,7 @@ func (ctlr *Controller) processResources() bool {
 		err := ctlr.processIngressLink(ingLink, rscDelete)
 		if err != nil {
 			// TODO
-			utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+			utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 			isRetryableError = true
 		}
 	case ExternalDNS:
@@ -446,7 +446,7 @@ func (ctlr *Controller) processResources() bool {
 				err := ctlr.processVirtualServers(virtual, false)
 				if err != nil {
 					// TODO
-					utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+					utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 					isRetryableError = true
 				}
 			}
@@ -456,7 +456,7 @@ func (ctlr *Controller) processResources() bool {
 				err := ctlr.processTransportServers(virtual, false)
 				if err != nil {
 					// TODO
-					utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+					utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 					isRetryableError = true
 				}
 			}
@@ -466,7 +466,7 @@ func (ctlr *Controller) processResources() bool {
 				err := ctlr.processLBServices(lbService, false)
 				if err != nil {
 					// TODO
-					utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+					utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 					isRetryableError = true
 				}
 			}
@@ -492,7 +492,7 @@ func (ctlr *Controller) processResources() bool {
 			err := ctlr.processLBServices(svc, rscDelete)
 			if err != nil {
 				// TODO
-				utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+				utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 				isRetryableError = true
 			}
 		}
@@ -531,7 +531,7 @@ func (ctlr *Controller) processResources() bool {
 			err := ctlr.processLBServices(svc, rscDelete)
 			if err != nil {
 				// TODO
-				utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+				utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 				isRetryableError = true
 			}
 		}
@@ -560,7 +560,7 @@ func (ctlr *Controller) processResources() bool {
 			err := ctlr.processLBServices(svc, rscDelete)
 			if err != nil {
 				// TODO
-				utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+				utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 				isRetryableError = true
 			}
 			break
@@ -592,14 +592,14 @@ func (ctlr *Controller) processResources() bool {
 				ctlr.namespacesMutex.Lock()
 				delete(ctlr.namespaces, nsName)
 				ctlr.namespacesMutex.Unlock()
-				log.Debugf("Removed Namespace: '%v' from CIS scope", nsName)
+				log.Infof("Removed Namespace: '%v' from CIS scope", nsName)
 				triggerDelete = true
 			} else {
 				ctlr.namespacesMutex.Lock()
 				ctlr.namespaces[nsName] = true
 				ctlr.namespacesMutex.Unlock()
 				_ = ctlr.addNamespacedInformers(nsName, true)
-				log.Debugf("Added Namespace: '%v' to CIS scope", nsName)
+				log.Infof("Added Namespace: '%v' to CIS scope", nsName)
 			}
 			if ctlr.namespaceLabelMode {
 				ctlr.processGlobalExtendedConfigMap()
@@ -615,7 +615,7 @@ func (ctlr *Controller) processResources() bool {
 					err := ctlr.processVirtualServers(vrt, true)
 					if err != nil {
 						// TODO
-						utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+						utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 						isRetryableError = true
 					}
 				}
@@ -624,7 +624,7 @@ func (ctlr *Controller) processResources() bool {
 					err := ctlr.processTransportServers(ts, true)
 					if err != nil {
 						// TODO
-						utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
+						utilruntime.HandleError(fmt.Errorf("[ERROR] Sync %v failed with %v", key, err))
 						isRetryableError = true
 					}
 				}
@@ -634,24 +634,29 @@ func (ctlr *Controller) processResources() bool {
 				ctlr.namespacesMutex.Lock()
 				delete(ctlr.namespaces, nsName)
 				ctlr.namespacesMutex.Unlock()
-				log.Debugf("Removed Namespace: '%v' from CIS scope", nsName)
+				log.Infof("Removed Namespace: '%v' from CIS scope", nsName)
 			} else {
 				ctlr.namespacesMutex.Lock()
 				ctlr.namespaces[nsName] = true
 				ctlr.namespacesMutex.Unlock()
 				_ = ctlr.addNamespacedInformers(nsName, true)
-				log.Debugf("Added Namespace: '%v' to CIS scope", nsName)
+				log.Infof("Added Namespace: '%v' to CIS scope", nsName)
 			}
 		}
 	case HACIS:
-		log.Debugf("posting declaration on primary cluster down event")
+		log.Infof("posting declaration on primary cluster down event")
 	case NodeUpdate:
-		log.Debugf("posting declaration on node update")
+		log.Infof("posting declaration on node update")
 	default:
 		log.Errorf("Unknown resource Kind: %v", rKey.kind)
 	}
 
 	if isRetryableError {
+		if rKey.clusterName == "" {
+			log.Warningf("Request from cluster local resulted in retry for  %v in %v %v/%v", strings.ToTitle(rKey.event), strings.ToTitle(rKey.kind), rKey.namespace, rKey.rscName)
+		} else {
+			log.Warningf("Request from cluster %v resulted in retry for %v in %v %v/%v", rKey.clusterName, strings.ToTitle(rKey.event), strings.ToTitle(rKey.kind), rKey.namespace, rKey.rscName)
+		}
 		ctlr.resourceQueue.AddRateLimited(key)
 	} else {
 		ctlr.resourceQueue.Forget(key)
@@ -684,6 +689,11 @@ func (ctlr *Controller) processResources() bool {
 			go ctlr.TeemData.PostTeemsData()
 		}
 		config.reqId = ctlr.enqueueReq(config)
+		if rKey.clusterName == "" {
+			log.Infof("[Request: %v] cluster local requested %v in %v %v/%v", config.reqId, strings.ToTitle(rKey.event), strings.ToTitle(rKey.kind), rKey.namespace, rKey.rscName)
+		} else {
+			log.Infof("[Request: %v] cluster %v requested %v in %v %v/%v", config.reqId, rKey.clusterName, strings.ToTitle(rKey.event), strings.ToTitle(rKey.kind), rKey.namespace, rKey.rscName)
+		}
 		ctlr.Agent.PostConfig(config)
 		ctlr.initState = false
 		ctlr.resources.updateCaches()
@@ -2083,7 +2093,7 @@ func (ctlr *Controller) fetchPoolMembersForService(serviceName string, serviceNa
 	}
 	err, svc := ctlr.fetchService(svcKey)
 	if err != nil {
-		log.Errorf("%v %v", err, getClusterLog(clusterName))
+		log.Warningf("service '%v' %s not found", svcKey, getClusterLog(clusterName))
 	}
 	var poolMembers []PoolMember
 	if svc != nil {
@@ -3913,7 +3923,7 @@ func (ctlr *Controller) processConfigMap(cm *v1.ConfigMap, isDelete bool) (error
 				}
 				ratioKeyValues += fmt.Sprintf(" %s:%d", cluster, *ratio)
 			}
-			log.Debugf("[MultiCluster] Cluster ratios:%s", ratioKeyValues)
+			log.Infof("[MultiCluster] Cluster ratios:%s", ratioKeyValues)
 		}
 		// Check if cluster Admin state has been updated for any cluster
 		// Check only if CIS is running in multiCluster mode

